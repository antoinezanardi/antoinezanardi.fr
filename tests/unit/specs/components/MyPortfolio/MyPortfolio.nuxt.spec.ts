import type { mount } from "@vue/test-utils";
import type { ComponentMountingOptions } from "@vue/test-utils/dist/mount";

import type MyProject from "~/components/MyPortfolio/MyProject/MyProject.vue";
import { MyPortfolio } from "#components";
import { mountSuspendedComponent } from "@tests/unit/utils/helpers/mount.helpers";
import type { Project } from "~/models/project/project.types";

describe("My Portfolio Component", () => {
  let wrapper: ReturnType<typeof mount<typeof MyPortfolio>>;

  async function mountMyPortfolioComponent(options: ComponentMountingOptions<typeof MyPortfolio> = {}): Promise<ReturnType<typeof mount<typeof MyPortfolio>>> {
    return mountSuspendedComponent(MyPortfolio, { ...options });
  }

  beforeEach(async() => {
    wrapper = await mountMyPortfolioComponent();
  });

  it("should match snapshot when rendered.", () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe("Projects", () => {
    it("should render 4 projects when rendered.", () => {
      const projects = wrapper.findAllComponents<typeof MyProject>(".project");

      expect(projects).toHaveLength(4);
    });

    it("should render my portfolio project when rendered.", () => {
      const projects = wrapper.findAllComponents<typeof MyProject>(".project");
      const expectedProject: Project = {
        name: "MyPortfolio.projects.portfolio.name",
        description: "MyPortfolio.projects.portfolio.description",
        image: "portfolio-thumbnail.jpg",
        url: "https://www.antoinezanardi.fr",
      };

      expect(projects[0].props("project")).toStrictEqual<Project>(expectedProject);
    });

    it("should render my Werewolves Assistant project when rendered.", () => {
      const projects = wrapper.findAllComponents<typeof MyProject>(".project");
      const expectedProject: Project = {
        name: "MyPortfolio.projects.werewolvesAssistant.name",
        description: "MyPortfolio.projects.werewolvesAssistant.description",
        image: "werewolves-assistant-thumbnail.png",
        url: "https://werewolves-assistant.com",
      };

      expect(projects[1].props("project")).toStrictEqual<Project>(expectedProject);
    });

    it("should render book distribution project when rendered.", () => {
      const projects = wrapper.findAllComponents<typeof MyProject>(".project");
      const expectedProject: Project = {
        name: "MyPortfolio.projects.distribution.name",
        description: "MyPortfolio.projects.distribution.description",
        image: "distribution-thumbnail.jpeg",
        url: "https://www.airvey-editions.fr",
      };

      expect(projects[2].props("project")).toStrictEqual<Project>(expectedProject);
    });

    it("should render my GitHub profile project when rendered.", () => {
      const projects = wrapper.findAllComponents<typeof MyProject>(".project");
      const expectedProject: Project = {
        name: "MyPortfolio.projects.gitHub.name",
        description: "MyPortfolio.projects.gitHub.description",
        image: "github.jpeg",
        url: "https://github.com/antoinezanardi",
      };

      expect(projects[3].props("project")).toStrictEqual<Project>(expectedProject);
    });
  });
});