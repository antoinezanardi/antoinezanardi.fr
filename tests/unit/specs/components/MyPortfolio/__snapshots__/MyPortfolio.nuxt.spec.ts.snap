// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`My Portfolio Component > should match snapshot when rendered. 1`] = `
"<div id="portfolio" aria-labelledby="portfolio-title" class="section" role="region">
  <div class="container">
    <section-title-stub icon="fa fa-star" title="Portfolio" iconcolor="#EEC624" id="portfolio-title"></section-title-stub>
    <div class="gallery tab-content">
      <div id="web-development" class="active tab-pane">
        <div class="me-auto ms-auto">
          <div class="align-items-center row">
            <div class="col-md-6">
              <my-project-stub project="[object Object]" class="project"></my-project-stub>
            </div>
            <div class="col-md-6">
              <my-project-stub project="[object Object]" class="project"></my-project-stub>
            </div>
            <div class="col-md-6">
              <my-project-stub project="[object Object]" class="project"></my-project-stub>
            </div>
            <div class="col-md-6">
              <my-project-stub project="[object Object]" class="project"></my-project-stub>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
